storage/fs_mgr.go

```diff
@@ -0,0 +1,57 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/m3db/m3x/clock"
+	"github.com/m3db/m3x/log"
+)
+
+type databaseFileSystemManager interface {
+	// Run attempts to flush the database if deemed necessary.
+	Run() bool
+}
+
+type runStatus int
+
+// nolint:deadcode,megacheck,varcheck
+const (
+	runNotStarted runStatus = iota
+	runInProgress
+)
+
+type fileSystemManager struct {
+	sync.Mutex
+
+	database database
+	opts     *Options
+	logger   log.Logger
+	nowFn    clock.NowFn
+
+	status runStatus
+}
+
+func newFileSystemManager(database database, opts *Options) *fileSystemManager {
+	return &fileSystemManager{
+		database: database,
+		opts:     opts,
+		logger:   opts.InstrumentOptions().Logger(),
+		nowFn:    opts.ClockOptions().NowFn(),
+	}
+}
+
+func (mgr *fileSystemManager) Run() bool {
```

Comment:
if this manager is only going to handle compaction, maybe call it compactionManager?

Reply:
There will also be file cleanup tasks (e.g., to delete expired data).